test/e2e/operator.go

```diff
@@ -379,33 +379,79 @@ var _ = Describe("ARO Operator - Azure Subnet Reconciler", func() {
 })
 
 var _ = Describe("ARO Operator - MUO Deployment", func() {
-	ctx := context.Background()
+	const (
+		managedUpgradeOperatorNamespace = "openshift-managed-upgrade-operator"
+		interval                        = 30 * time.Second
+		duration                        = 10 * time.Minute
+	)
+
+	var ctx context.Context
+
+	muoIsDeployed := func() (bool, error) {
```

Comment:
I believe this should be part of beforeEach, and the checks for `mandated: true` can be refactored out into the first test below. Right now you are testing the same case twice in both tests (the second one checking more things afterwards) and also listing the pod twice. This can be done once as a before step. It is actually good practice to even have assertions for correct setups in your beforeEach step.

Reply:
by "also listing the pod twice" I mean duplicating the code

Reply:
Refactored to avoid duplication. Thanks!